exonum/src/sandbox/sandbox.rs

```diff
@@ -743,13 +747,30 @@ impl SandboxBuilder {
         self
     }
 
+    pub fn with_validators(mut self, n: u8) -> Self {
+        self.validators_count = n;
+        self
+    }
+
+    pub fn with_logger(mut self) -> Self {
+        self.logger = true;
+        self
+    }
+
     pub fn build(self) -> Sandbox {
-        let mut sandbox = sandbox_with_services_uninitialized(self.services, self.consensus_config);
+        if self.logger {
+            let _ = env_logger::try_init();
```

Comment:
Logger initialisation causes side effects: other test inside this process will get init logger as well.

Reply:
Yup. But that won't affect anything.
That's exactly why we use `try_init`

Reply:
Hm... I see, but is there a way to avoid that?

Reply:
Uh... Why do we need the logging to be configurable per-test? Maybe we can always initialize logging (probably will need to use `std::sync::Once` to do it only once) and then just fiddle with `RUST_LOG` environment variable when we run particular tests during debugging. I do not think that log output by default is valuable: if the test passes then I don't really care what happens there.

Reply:
@ilammy 
> Why do we need the logging to be configurable per-test?
Have no idea. It's the same scheme we use in TestKit. 
>  will need to use std::sync::Once to do it only once
`try_init` will init it only once anyway.

Reply:
Right! I remembered what would be nice to have.

For tests we should probably configure the logger to write output to **stdout** instead of stderr. This makes test log more readable as successful tests will not output anything other than "ok" while failing tests will dump the log along with panics and assertion failures.

Reply:
`2>&1` redirects stderr to stdout
`2>errrors.log` writes it to file

Reply:
@ponimas there a big difference between "output log to stdout" and "redirect log to stdout".
You redirect log outside of current process.

Outputting log into stderr allows libtest to capturing, and only showing it on test fail.

So, i agrue with @liammy, can we:
1) init log using https://docs.rs/env_logger/0.5.12/env_logger/struct.Builder.html (with target Stdout)
2) initi it unconditionally and remove `with_logger` from sandbox builder 
3) probably use `Once` to init logger with builder

Reply:
@vldm  ok, now I see the point.
But have 2 questions left:
- Does that mean that we should do the same in the TestKit?
- I still can't get why the hell we should use `Once` and can't just use `try_init` and will be grateful for an explanation.

Reply:
> Does that mean that we should do the same in the TestKit?
Probably yes, based on logging purposes in testkit @slowli 

> I still can't get why the hell we should use Once and can't just use try_init and will be grateful for an explanation.

https://docs.rs/env_logger/0.5.12/src/env_logger/lib.rs.html#725-727

`builder.init()` will fail at second initialisation